lib/reversetunnel/agentpool.go

```diff
@@ -67,7 +67,7 @@ const (
 // lib/srv in lib/reversetunnel causes a circular import.
 type ServerHandler interface {
 	// HandleConnection performs a handshake then process the connection.
-	HandleConnection(conn net.Conn)
+	HandleConnection(conn net.Conn, targetPort int)
```

Comment:
In a real implementation, I'd use some kind of a struct to pass this extra data, not just a plain int arg. This extra piece of data needs to be accepted by all server handlers, but it's only really consumed by the app service.